client/src/components/BlockSearch.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+<div>
+  <q-virtual-scroll
+    class="virtual-scroll"
+    :items="results"
+  >
+    <template v-slot="{ item: user, index }">
+      <q-item
+        :key="index"
+        dense
+        clickable
+        v-if="!results[0].error"
+      >
+        <q-item-section avatar>
+          <q-avatar>
+            <avatar size="40px" :img="user.avatar" />
+          </q-avatar>
+        </q-item-section>
+        <q-item-section class="col" @click="$emit('handleChoose', user.id, user.nickname)">{{ user.nickname }}</q-item-section>
+      </q-item>
+      <q-item v-else>
+        <q-item-section>
+          <p>{{ results[0].error }}</p>
+        </q-item-section>
+      </q-item>
+    </template>
+  </q-virtual-scroll>
+</div>
+</template>
+
+<script>
+import Avatar from '@/components/Avatar.vue'
+export default {
+  components: {
+    Avatar
+  },
+  data () {
+    return {}
+  },
+  props: ['results']
```

Comment:
As improvement you could describe props type, for example, props: { results: { type: Array, default: () => {//some default value} } }